Zotero/Models/Database/Database.swift

```diff
@@ -50,9 +50,26 @@ struct Database {
             if schemaVersion < 39 {
                 self.addUuidsToCreators(migration: migration)
             }
+            if schemaVersion < 40 {
+                truncateAnnotationPageLabels(migration: migration)
+            }
         }
     }
 
+    private static func truncateAnnotationPageLabels(migration: Migration) {
+        migration.enumerateObjects(ofType: RItem.className()) { oldObject, newObject in
+            let maxLength = 16
+            guard oldObject?["rawType"] as? String == ItemTypes.annotation,
+                  var fields = oldObject?["fields"] as? List<MigrationObject>,
+                  let index = fields.firstIndex(where: { $0["key"] as? String == FieldKeys.Item.Annotation.pageLabel }),
+                  let pageLabel = fields[index]["value"] as? String,
+                  pageLabel.count > maxLength
+            else { return }
+            (newObject?["fields"] as? List<MigrationObject>)?[index]["value"] = pageLabel.prefix(maxLength)
+            (newObject?["fields"] as? List<MigrationObject>)?[index]["changed"] = true
```

Comment:
If we want this to sync, you need to add a `RObjectChange` with value of `RItemChanges.fields` for this `RItem` too.

Reply:
Check https://github.com/zotero/zotero-ios/pull/784/commits/3649765e8a45246a2910e20b07f99fc6c62613e4